### PR TITLE
User shouldn't be able to approve or reject his/her own PR

### LINKED ISSUE #4728: User can approve or reject his/her own PR
- Gitea version (or commit ref): bc06ab4a31c54c5d164d53ec16d5c1089be9c3fe
- Git version: 2.15.2 (Apple Git-101.1)
- Operating system: MacOS
- Database (use `[x]`): 
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes ( https://try.gitea.io/lanre.adelowo/dotfiles/pulls/1#issuecomment-7378 )
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

A user can approve and/or reject a PR (s)he created... This behaviour doesn't exist on Github for example

> PS : (s)he can still merge the PR but cannot leave an approval or rejection review

...

screen shot 2018-08-16 at 15 27 51
screen shot 2018-08-16 at 15 27 54

### PR DESCRIPTION
**Pull request by adelowo**
Fixes #4728

### CODE REVIEW

File: routers/repo/pull.go [hunk 1]
```diff
@@ -484,6 +484,7 @@ func ViewPullFiles(ctx *context.Context) {
 		return
 	}
 
+	ctx.Data["IsOwner"] = ctx.Repo.IsWriter() || (ctx.IsSigned && issue.IsPoster(ctx.User.ID))
```
Comment (adelowo) [inline routers/repo/pull.go]: Should this just be `ctx.IsSigned && issue.IsPoster(ctx.User.ID)`

**Comment by lafriks** [comment]
Do not create new template variables if possible

**Comment by adelowo** [comment]
~~I’m currently in transit and would look at it again to see if that would be possible @lafriks~~

@lafriks I don't think it can be done without adding a new template variable though

**Comment by lafriks** [comment]
You can try if this works `{{ if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID) }}` (have not tested it myself)

File: routers/repo/pull_review.go [hunk 2]
```diff
@@ -103,9 +103,29 @@ func SubmitReview(ctx *context.Context, form auth.SubmitReviewForm) {
...
+			var translated string
+
+			if reviewType == models.ReviewTypeApprove {
+				translated = ctx.Tr("repo.issues.review.self.approval")
+			} else {
+				translated = ctx.Tr("repo.issues.review.self.rejection")
+			}
+
+			ctx.Flash.Error(translated)
+			ctx.Redirect(fmt.Sprintf("%s/pulls/%d", ctx.Repo.RepoLink, issue.Index))
```
Comment (jonasfranz) [inline routers/repo/pull_review.go]: Please redirect to `%s/pulls/%d/files` instead.

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: routers/repo/pull_review.go (hunk 2 above)
Comment (adelowo) [reply routers/repo/pull_review.go]: The `base/alert` template isn't imported in the template for the page `%s/pulls/%d/files` which is why I redirected back to the main page... Should I just add the alert template in there or just leave it as it is ? ( #4632  redirects to the main PR page though )

Comment (lafriks) [reply routers/repo/pull_review.go]: Add alert to to files template

Comment (adelowo) [reply routers/repo/pull_review.go]: Fixed @JonasFranzDEV and @lafriks

**Review by lafriks — APPROVED** [review]

**Review by jonasfranz — APPROVED** [review]

**Pull request merged**